### PR TITLE
Add a config to decode json-encoded strings in attribute values

### PR DESCRIPTION
**Pull request by pxaws**
**Description:** <Describe what has changed.>

This commit adds a config `parse_json_encoded_attr_values` which is the list of attributes whose values are json-encoded strings.  The emf exporter will convert those json-encoded strings as json structures before putting them into the emf log events. 

This is to support the aws container insight use case in which json objects are used as field values. For example,  like the `kubernetes` and `Sources` field in the following emf log event:
```
{
    "AutoScalingGroupName": "eks-a6bb9db9-267c-401c-db55-df8ef645b06f",
    "ClusterName": "ci-demo",
    "InstanceId": "i-01bf9fb097cbf3205",
    "InstanceType": "t2.xlarge",
    "Namespace": "aws-otel-eks",
    "NodeName": "ip-192-168-12-170.ec2.internal",
    "PodName": "aws-otel-eks-ci",
    "Sources": [
        "cadvisor",
        "calculated"
    ],
    "Type": "PodNet",
    "Version": "0",
    "interface": "eth0",
    "kubernetes": {
        "host": "ip-192-168-12-170.ec2.internal",
        "labels": {
            "controller-revision-hash": "66bb8fc7f4",
            "name": "aws-otel-eks-ci",
            "pod-template-generation": "1"
        },
        "namespace_name": "aws-otel-eks",
        "pod_id": "1810eef4-9121-4e93-8b0a-020c54f7924b",
        "pod_name": "aws-otel-eks-ci-xp66r",
        "pod_owners": [
            {
                "owner_kind": "DaemonSet",
                "owner_name": "aws-otel-eks-ci"
            }
        ]
    },
    "pod_interface_network_rx_bytes": 3166.8867430387045,
    "pod_interface_network_rx_dropped": 0,
    "pod_interface_network_rx_errors": 0,
    "pod_interface_network_rx_packets": 3.7814973119633524,
    "pod_interface_network_total_bytes": 4809.434331265391,
    "pod_interface_network_tx_bytes": 1642.5475882266862,
    "pod_interface_network_tx_dropped": 0,
    "pod_interface_network_tx_errors": 0,
    "pod_interface_network_tx_packets": 2.990021130389628
}
```
Since the api `metric.LabelsMap()` in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/awsemfexporter/datapoint.go#L106 returns a key value string map, a json object like this
```
{
        "host": "ip-192-168-12-170.ec2.internal",
        "labels": {
            "controller-revision-hash": "66bb8fc7f4",
            "name": "aws-otel-eks-ci",
            "pod-template-generation": "1"
        },
        "namespace_name": "aws-otel-eks",
        "pod_id": "1810eef4-9121-4e93-8b0a-020c54f7924b",
        "pod_name": "aws-otel-eks-ci-xp66r",
        "pod_owners": [
            {
                "owner_kind": "DaemonSet",
                "owner_name": "aws-otel-eks-ci"
            }
        ]
    }
```
can't be used directly. To work around the limitation, we stringify the json value and save it as string attribute value in receiver and then restore the original json in emf exporter.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2307

**Testing:** <Describe what testing was performed and which tests were added.>
Updated the unit tests.

**Documentation:** <Describe the documentation added.>
Updated the readme to include the new config.

### CODE REVIEW

File: exporter/awsemfexporter/metric_translator.go [hunk 1]
```diff
@@ -332,11 +332,24 @@ func groupedMetricToCWMeasurementsWithFilters(groupedMetric *GroupedMetric, conf
...
 
+	//restore the json objects that are stored as string in attributes
+	for _, key := range config.ParseJSONEncodedAttributeValues {
+		if fieldMap[key] == nil {
+			continue
+		}
+		var f interface{}
+		err := json.Unmarshal([]byte(fieldMap[key].(string)), &f)
+		if err != nil {
+			continue
```
Comment (mxiamxia) [inline exporter/awsemfexporter/metric_translator.go:348]: Add log here for debugging?

Comment (pxaws) [reply exporter/awsemfexporter/metric_translator.go:365]: Sure.

File: exporter/awsemfexporter/metric_translator.go [hunk 2]
```diff
@@ -332,11 +333,40 @@ func groupedMetricToCWMeasurementsWithFilters(groupedMetric *GroupedMetric, conf
...
 	cWMetricMap := make(map[string]interface{})
 	fieldMap := cWMetric.Fields
 
+	//restore the json objects that are stored as string in attributes
+	for _, key := range config.ParseJSONEncodedAttributeValues {
+		if fieldMap[key] == nil {
+			continue
+		}
+
+		if t := reflect.TypeOf(fieldMap[key]).Name(); t != "string" {
```
Comment (anuraaga) [inline exporter/awsemfexporter/metric_translator.go:347]: I think you can do something like `if t, ok := t.(string); ok`

Comment (pxaws) [reply exporter/awsemfexporter/metric_translator.go:347]: Cool. This is better.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**